src/sinks/gcp/stackdriver_metrics.rs

```diff
@@ -0,0 +1,215 @@
+use crate::config::{DataType, SinkConfig, SinkContext, SinkDescription};
+use crate::event::{Metric, MetricValue};
+use crate::http::HttpClient;
+use crate::sinks::gcp;
+use crate::sinks::util::buffer::metrics::MetricsBuffer;
+use crate::sinks::util::http::{BatchedHttpSink, HttpSink};
+use crate::sinks::util::{BatchConfig, BatchSettings, TowerRequestConfig};
+use crate::sinks::{Healthcheck, VectorSink};
+use crate::tls::{TlsOptions, TlsSettings};
+use crate::Event;
+use chrono::{DateTime, Utc};
+use futures::sink::SinkExt;
+use futures::FutureExt;
+use http::header::AUTHORIZATION;
+use http::{HeaderValue, Uri};
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct StackdriverConfig {
+    pub project_id: String,
+    pub resource: gcp::GcpMonitoredResource,
+    pub credentials_path: Option<String>,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+    #[serde(default)]
+    pub batch: BatchConfig,
+    pub tls: Option<TlsOptions>,
+}
+
+impl_generate_config_from_default!(StackdriverConfig);
+
+inventory::submit! {
+    SinkDescription::new::<StackdriverConfig>("gcp_stackdriver_metrics")
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "gcp_stackdriver_metrics")]
+impl SinkConfig for StackdriverConfig {
+    async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let mut token = gouth::Builder::new().scopes(&[
+            "https://www.googleapis.com/auth/cloud-platform",
+            "https://www.googleapis.com/auth/monitoring",
+            "https://www.googleapis.com/auth/monitoring.write",
+        ]);
+
+        if let Some(credentials_path) = self.credentials_path.clone() {
+            token = token.file(credentials_path);
+        }
+
+        let token = token.build()?;
```

Comment:
```suggestion
        let token = self
            .credentials_path
            .as_ref()
            .map(|path| token.file(path))
            .unwrap_or(token)
            .build()?;
```

Reply:
This will not compile considering token is moved at line 61.